plugins/single_plugins/ipc-rules.cpp

```diff
@@ -333,6 +340,46 @@ class ipc_rules_t : public wf::plugin_interface_t, public wf::per_output_tracker
         return wf::ipc::json_ok();
     };
 
+    nlohmann::json wset_to_json(wf::workspace_set_t *wset)
+    {
+        nlohmann::json response;
+        response["id"]   = wset->get_id();
+        response["name"] = wset->to_string();
+
+        auto output = wset->get_attached_output();
+        response["output-id"]   = output ? output->get_id() : -1;
```

Comment:
`get_id()` returns a uint, so we need to cast it explicitly to `int` to avoid overflow for the -1. There are also other places where the exact same bug happens, I will fix them as well at some point, but we shouldn't introduce even more bugs ;)